src/rpcdiscovery.rs

```diff
@@ -0,0 +1,215 @@
+use shvproto::RpcValue;
+
+use crate::metamethod::{AccessLevel, DirAttribute};
+
+#[derive(Debug, PartialEq)]
+pub enum DirParam {
+    Brief,
+    Full,
+    BriefMethod(String),
+}
+
+impl From<&RpcValue> for DirParam {
+    fn from(rpcvalue: &RpcValue) -> Self {
+        match rpcvalue.value() {
+            shvproto::Value::String(param) => DirParam::BriefMethod(param.to_string()),
+            shvproto::Value::Bool(true) => DirParam::Full,
+            _ => DirParam::Brief,
+        }
+
+    }
+}
+
+impl From<Option<&RpcValue>> for DirParam {
+    fn from(value: Option<&RpcValue>) -> Self {
+        match value {
+            Some(rpcval) => rpcval.into(),
+            None => DirParam::Brief,
+        }
+    }
+}
+
+impl From<DirParam> for RpcValue {
+    fn from(value: DirParam) -> Self {
+        match value {
+            DirParam::Brief => false.into(),
+            DirParam::Full => true.into(),
+            DirParam::BriefMethod(method) => method.into(),
+        }
+    }
+}
+
+
+#[derive(Debug,PartialEq)]
+pub struct MethodInfo {
+    pub name: String,
+    pub flags: u32,
+    pub access: AccessLevel,
```

Comment:
`accessLevel` please, previous SHV RPC versions had also string `access` field, which can be confused with access level if it has the same name.

Reply:
I have also found out that there are implementations of `dir` that return `"accessGrant"` as a key, so I added also this case to the conversion method.